code/modules/modular_computers/file_system/programs/robocontrol.dm

```diff
@@ -35,20 +35,37 @@
 	botcount = 0
 	current_user = user
 
-	for(var/B in GLOB.bots_list)
-		var/mob/living/simple_animal/bot/Bot = B
-		if(!(Bot.bot_mode_flags & BOT_MODE_ON) || Bot.z != zlevel || !(Bot.bot_mode_flags & BOT_MODE_REMOTE_ENABLED)) //Only non-emagged bots on the same Z-level are detected!
+	for(var/mob/living/simple_animal/bot/Bot as anything in GLOB.bots_list)
+		if(Bot.z != zlevel || !(Bot.bot_mode_flags & BOT_MODE_REMOTE_ENABLED)) //Only non-emagged bots on the same Z-level are detected!
 			continue
 		else if(computer) //Also, the inserted ID must have access to the bot type
 			var/obj/item/card/id/id_card = card_slot ? card_slot.stored_card : null
-			if(!id_card && !Bot.bot_core.allowed(current_user))
+			if(!id_card && !Bot.check_access(current_user))
```

Comment:
You're not passing in the card at all here.

Reply:
it wasnt needed anymore so I just removed that now